### PR TITLE
Use twisted.logger for logging errors in mailmail.py

### PR DESCRIPTION
**Pull request by rodrigc**
Moved from:
https://github.com/twisted/twisted/pull/200

See:
https://twistedmatrix.com/trac/ticket/8487

### CODE REVIEW

File: twisted/mail/test/test_mailmail.py [hunk 1]
```diff
@@ -73,3 +100,129 @@ def test_copyAliasedSender(self):
...
+        o = parseOptions(["-F", "Groucho <invaliduser5@example.com>", "-t"])
+        self.assertEqual(o.sender, "invaliduser4@example.com")
+
+
+    def test_run(self):
+        """
+        Test run() method.
+        """
+        if platformType == "win32":
+            raise SkipTest("win32 lacks support for getuid()")
```
Comment (adiroiban) [inline twisted/mail/test/test_mailmail.py]: this is strange. it looks like platformType is never 'win32' as this line is never executed in the test.

I think that in Twisted, tests should be skipped using `test_run.skip = "win32 lacks support for getuid()"`

File: twisted/mail/scripts/mailmail.py [hunk 2]
```diff
@@ -260,7 +262,9 @@ def loadConfig(path):
                         try:
                             id = int(id)
                         except ValueError:
-                            log("Illegal %sID in [%s] section: %s", section[0].upper(), section, id)
+                            log.error(
+                                "Illegal {}ID in [{}] section: {}".format(
+                                section[0].upper(), section, id))
```
Comment (adiroiban) [inline twisted/mail/scripts/mailmail.py]: Please use the structured log call.

ex

```
log.info(
    "{request.uri}: status={response.status}, bytes={response.size}, etc...",
    request=request, response=response
)
```

For more details see 'The basics' section from the logger docs https://twistedmatrix.com/documents/current/core/howto/logger.html#the-basics

Already serialized strings for the logging systesm are so '90 :)

File: twisted/mail/test/test_mailmail.py [hunk 3]
```diff
@@ -73,3 +100,129 @@ def test_copyAliasedSender(self):
...
+        data = out.getvalue()
+        self.assertEqual(data, "mailmail version: {}\n".format(version))
+
+
+    def test_backGroundDelivery(self):
+        """
+        The I{-odb} flag specifies background delivery.
+        """
+        self.addCleanup(setattr, sys, 'stdin', sys.stdin)
+        sys.stdin = StringIO('\n')
```
Comment (adiroiban) [inline twisted/mail/test/test_mailmail.py]: why not use self.patch here?

File: twisted/mail/scripts/mailmail.py [hunk 4]
```diff
@@ -274,7 +280,7 @@ def loadConfig(path):
         for (host, up) in p.items('identity'):
             parts = up.split(':', 1)
             if len(parts) != 2:
-                log("Illegal entry in [identity] section: %s", up)
+                log.error("Illegal entry in [identity] section: {}".format(up))
```
Comment (wsanchez) [inline twisted/mail/scripts/mailmail.py]: Don't call format(), just add a kwarg.

**Pull request closed without merging**